lnwallet/channel.go

```diff
@@ -1765,6 +1765,12 @@ func (lc *LightningChannel) restoreStateLogs(
 		}
 
 		if payDesc.EntryType == Add {
+			if payDesc.HtlcIndex != lc.localUpdateLog.htlcCounter {
```

Comment:
Why is this necessary?

Reply:
This was added as `appendHtlc` gives the HTLC an index based on update log's `htlcCounter`. If the set `HtlcIndex` is different from this counter then it will cause problems.

I will make this a panic and add a TODO to remove it when the cause of the bug is found.